SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -979,6 +996,34 @@ void ActorSystem::ApplyComponentData(USpatialActorChannel& Channel, UObject& Tar
 	}
 }
 
+void ActorSystem::ResolvePendingOpsFromEntityUpdate(TArray<FWeakObjectPtr> ToResolveOps)
+{
+	// This should be called after all component updates and adds have been completed, and PostNetReceives have been called to avoid user
+	// code from seeing inconsistent state
+	for (const FWeakObjectPtr& WeakObjectPtr : ToResolveOps)
+	{
+		UObject* Object = WeakObjectPtr.Get();
+		if (!Object)
+		{
+			UE_LOG(LogActorSystem, Log,
+				   TEXT("ResolvePendingOpsFromEntityUpdate: Did not resolve pending ops for object %s as it was no longer valid."),
+				   *GetNameSafe(Object));
+			continue;
+		}
+
+		const FUnrealObjectRef ObjectRef = NetDriver->PackageMap->GetUnrealObjectRefFromObject(Object);
+
+		if (!ObjectRef.IsValid())
+		{
+			UE_LOG(LogActorSystem, Error,
+				   TEXT("ResolvePendingOpsFromEntityUpdate: Tried to resolve pending ops for %s but object ref was not valid."),
+				   *Object->GetName());
+		}
+
+		ResolvePendingOperations(Object, ObjectRef);
+	}
+}
```

Comment:
I don't think it really matters, and is perhaps good, if we don't resolve a user object reference because the object was pendingkill (e.g. it was killed in a Pre or PostNetReceive), but not sure about RPCs - @ImprobableNic any thoughts?